bonus_guides/D_mix_tasks.md

```diff
@@ -810,3 +810,14 @@ Greetings from the Hello Phoenix Application!
 ```
 
 Indeed it does.
+
+If you want to make your new mix task to use your application's infrastructure, you need to make sure the application is started when mix task is being executed. This is particulary useful if you need to access your database from within the mix task. Thankfully, mix makes it really easy for us:
+
+```elixir
+  . . .
+  def run(_args) do
+    Mix.Task.run "app.start"
```

Comment:
If inside a mix task and you want to start your app it's better to call you `Mix.Task.run "app.start"` than `Application.ensure_started/1`.

In many cases it's easier to just start your app instead of starting specific parts like the repo because they may depend on each other and app start is usually very quick anyway.